common/src/main/java/org/opensearch/ml/common/breaker/MLCircuitBreakerService.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.common.breaker;
+
+import lombok.extern.log4j.Log4j2;
+import org.opensearch.monitor.jvm.JvmService;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+/**
+ * This service registers internal system breakers and provide API for users to register their own breakers.
+ */
+@Log4j2
+public class MLCircuitBreakerService {
+
+    private final ConcurrentMap<BreakerName, CircuitBreaker> breakers = new ConcurrentHashMap<>();
+    private final JvmService jvmService;
+
+    /**
+     * Constructor.
+     *
+     * @param jvmService jvm info
+     */
+    public MLCircuitBreakerService(JvmService jvmService) {
+        this.jvmService = jvmService;
+    }
+
+    public void registerBreaker(BreakerName name, CircuitBreaker breaker) {
+        breakers.putIfAbsent(name, breaker);
+    }
+
+    public void unregisterBreaker(BreakerName name) {
+        if (name == null) {
+            return;
+        }
+
+        breakers.remove(name);
+    }
```

Comment:
Wondering if a log can be added here just like the log message in registerBreaker().

Reply:
Sure, will add in next PR.